### PR TITLE
Replace current install variables with GNUInstallDirs

### PR DESCRIPTION
**Pull request by SoapGentoo**
* The GNUInstallDirs module is more idiomatic and supported by
  Kitware upstream, whereas the current directories are not
  standardised across CMake-using packages. Using CMake native
  mechanisms is better than reinventing the wheel, as it makes
  using the build system more uniform across the ecosystem
* Use CMAKE_CXX_STANDARD to force C++11
* Require CMake 3.1.0 at a minimum
* Fixed lower/UPPERcase format for function/macro calls
* Fixed indents by replacing tabs with 4 spaces

This PR is in response to https://github.com/open-source-parsers/jsoncpp/issues/560

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -86,65 +76,71 @@ CONFIGURE_FILE( "${PROJECT_SOURCE_DIR}/version.in"
...
-if(CCACHE_FOUND)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
-endif(CCACHE_FOUND)
+    SET(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /W4 ")
+ENDIF()
+
+# Require C++11 support, prefer ISO C++ over GNU variants,
+# as relying solely on ISO C++ is more portable.
+SET(CMAKE_CXX_STANDARD 11)
```
Comment (strahlc) [inline CMakeLists.txt]: this feature requires at least cmake 3.1:

You should change line 3 to:
cmake_minimum_required(VERSION 3.1)

Comment (SoapGentoo) [inline CMakeLists.txt]: Done!

**Comment by cdunn2001** [comment]
Can't merge if Travis fails. Maybe there is a way to load the right cmake?

**Comment by SoapGentoo** [comment]
@cdunn2001 I've fixed the CMake error by switching to a newer CMake version, using https://github.com/travis-ci/travis-ci/issues/4631#issuecomment-191153634
Would you consider merging now? :smile:

**Comment by cdunn2001** [comment]
This looks like a lot of work. Thanks!

**Comment by cdunn2001** [comment]
I have this error:
```
$ cmake -DCMAKE_BUILD_TYPE=debug -DBUILD_STATIC_LIBS=ON -DBUILD_SHARED_LIBS=OFF -DARCHIVE_INSTALL_DIR=. -G "Unix Makefiles" ../..
CMake Error at CMakeLists.txt:4 (PROJECT):
  The CMAKE_C_COMPILER:
    /Library/Developer/CommandLineTools/usr/bin/cc

  is not a full path to an existing compiler tool.
```
I'll try setting `CC` in my environment.  I guess we need to update the README. Would you do that?

**Comment by cdunn2001** [comment]
Oh, it worked after removing the cmake cache. (Actually, I removed the whole build directory and started over.) No problems!

**Comment by cdunn2001** [comment]
https://github.com/open-source-parsers/jsoncpp/tree/1.8.0

**Pull request merged**